### ISSUE TITLE
Crash when sending a big response

### ISSUE THREAD

**Issue by ktosiek** [FreeRADIUS/freeradius-server#1569]
# Issue type
Defect - Crash or memory corruption.

I have FR configured to load some additional attributes from MSSQL (multiple values for Filter-id). I see a crash when user has a lot of those attributes.
It seems it crashes just before sending the packet - I'm still getting the "Auth OK" line in logs.
This system is disconnected, so I don't have all the data requested in the issue template, I'll fill them when I get access to them.

# Defect/Feature description
## How to reproduce issue
Get a lot of attributes from MSSQL and try to respond with Access-Accept.
 
## Output of ``[radiusd|freeradius] -X`` showing issue ocurring
Don't have it yet, I'll try to get to it shortly.

## Full backtrace from LLDB or GDB
```
(gdb) info threads
  Id   Target Id         Frame 
* 1    Thread 0x7fb5ca055780 (LWP 2722) rad_sign (packet=packet@entry=0x352e3130312e3032, original=original@entry=0xd6e335, secret=secret@entry=0xcdd610 "testing123") at src/lib/radius.c:1922
(gdb) thread apply all bt full

Thread 1 (Thread 0x7fb5ca055780 (LWP 2722)):
#0  rad_sign (packet=packet@entry=0x352e3130312e3032, original=original@entry=0xd6e335, secret=secret@entry=0xcdd610 "testing123") at src/lib/radius.c:1922
        hdr = <optimized out>
#1  0x00007fb5c99f9c40 in rad_send (packet=0x352e3130312e3032, original=0xd6e335, secret=0xcdd610 "testing123") at src/lib/radius.c:2056
No locals.
#2  0x0000000000419edd in auth_socket_send (listener=<optimized out>, request=0xd6e4d0) at src/main/listen.c:1318
No locals.
#3  0x0000000000433320 in request_finish (request=0xd6e4d0, action=<optimized out>) at src/main/process.c:1462
        vp = <optimized out>
#4  0x000000000042fe44 in request_queue_or_run (request=request@entry=0xd6e4d0, process=process@entry=0x433ae0 <request_running>) at src/main/process.c:1015
No locals.
#5  0x0000000000431527 in request_receive (ctx=ctx@entry=0xd6e2e0, listener=listener@entry=0xd6bb00, packet=0xd6e340, client=client@entry=0xcdd320, fun=fun@entry=0x410480 <rad_authenticate>) at src/main/process.c:1782
        count = <optimized out>
        packet_p = <optimized out>
        request = <optimized out>
        now = {tv_sec = 1458660782, tv_usec = 414770}
        sock = <optimized out>
#6  0x000000000041bdfe in auth_socket_recv (listener=0xd6bb00) at src/main/listen.c:1568
        rcode = <optimized out>
        code = 1
        src_port = 37558
        packet = <optimized out>
        fun = 0x410480 <rad_authenticate>
        client = 0xcdd320
        src_ipaddr = {af = 2, ipaddr = {ip4addr = {s_addr = 16777343}, ip6addr = {__in6_u = {__u6_addr8 = "\177\000\000\001", '\000' <repeats 11 times>, __u6_addr16 = {127, 256, 0, 0, 0, 0, 0, 0}, __u6_addr32 = {16777343, 0, 0, 
                  0}}}}, prefix = 32 ' ', scope = 0}
        ctx = 0xd6e2e0
#7  0x000000000042e4be in event_socket_handler (xel=<optimized out>, fd=<optimized out>, ctx=<optimized out>) at src/main/process.c:4556
        listener = <optimized out>
#8  0x00007fb5c9a02805 in fr_event_loop (el=0xce0bf0) at src/lib/event.c:641
        ef = 0xce0c40
        i = 1
        rcode = <optimized out>
        when = {tv_sec = 0, tv_usec = 0}
        wake = <optimized out>
        maxfd = 13
        read_fds = {fds_bits = {128, 0 <repeats 15 times>}}
        master_fds = {fds_bits = {16288, 0 <repeats 15 times>}}
#9  0x00000000004350f1 in radius_event_process () at src/main/process.c:5621
No locals.
#10 0x000000000040f776 in main (argc=<optimized out>, argv=0x7fff142b3268) at src/main/radiusd.c:579
        rcode = 0
        status = <optimized out>
        argval = <optimized out>
        spawn_flag = false
        display_version = false
        flag = 0
        from_child = {-1, -1}
        state = 0x667920 <global_state>
        autofree = 0xa88070
```

**Comment by alandekok** [comment]
Can you try with the v3.0.x head? I think this was found and fixed in some commits in January.

**Comment by ktosiek** [comment]
Thanks, I'll try to reproduce it in a test environment tomorrow, and try the new version there.

Dnia 22 marca 2016 19:20:09 CET, Alan DeKok <notifications@github.com> napisał(a):
>Can you try with the v3.0.x head? I think this was found and fixed in
>some commits in January.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/FreeRADIUS/freeradius-server/issues/1569#issuecomment-199949412

-- 
Wysłane za pomocą K-9 Mail.

**Comment by ktosiek** [comment]
Looks like it was fixed somewhere between release_3_0_11 and ce634c031a783ac0bbcfbdda9b006689a09b0eee. How safe is v3.0.x to run in production? Should I find and cherry-pick the fix, or is it better to go with unreleased version?

**Comment by alandekok** [comment]
We will be releasing 3.0.12 soon, from the v3.0.x branch.  It's fine to run in production.

All new development has been moved to the v3.1.x branch.  The v3.0.x branch is only for minor bug fixes at this point.
